codegen-core/src/test/kotlin/software/amazon/smithy/rust/codegen/core/smithy/generators/InstantiatorTest.kt

```diff
@@ -91,7 +91,9 @@ class InstantiatorTest {
     // This is the exact same behavior of the client.
     private class BuilderKindBehavior(val codegenContext: CodegenContext) : Instantiator.BuilderKindBehavior {
         override fun hasFallibleBuilder(shape: StructureShape) =
-            BuilderGenerator.hasFallibleBuilder(shape, codegenContext.symbolProvider)
+            // ClientBuilderGenerator.hasFallibleBuilder(shape, codegenContext.symbolProvider)
+            // TODO: Not sure how to fix this. Maybe this module should be in codegen-client too?
+            false
```

Comment:
You can take a look at the event stream tests to see how I abstracted out the builder generator so that it can be used by both the client and server tests. Basically, the common test code is in `codegen-core` in `EventStreamTestTools.kt`, and then there are Client/Server test cases (e.g., `ServerEventStreamMarshallerGeneratorTest.kt`) in the `codegen-client`/`codegen-server` modules that run the actual test cases using the correct builders.

Reply:
I ended up moving these tests directly to `codegen-client`. They literally used a copy-pasted version of `ClientBuilderKindBehavior` so they were pretty much just more `ClientInstantiator` tests.